client/src/main/java/io/split/client/LocalhostSplitFile.java

```diff
@@ -0,0 +1,133 @@
+                if (key == null) {
+                    Thread.yield();
+                    continue;
+                }
+
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    WatchEvent.Kind<?> kind = event.kind();
+
+                    @SuppressWarnings("unchecked")
+                    WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                    Path filename = ev.context();
+
+                    if (kind == StandardWatchEventKinds.OVERFLOW) {
+                        Thread.yield();
+                        continue;
+                    } else if (kind == java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY
+                            && filename.toString().equals(_file.getName())) {
+                        Map<String, String> featureToSplitMap = readOnSplits();
+                        _splitFactory.updateFeatureToTreatmentMap(featureToSplitMap);
+                        _log.info("Detected change in Local Splits file - Splits Reloaded! file={}", _file.getPath());
+                    }
+                    boolean valid = key.reset();
+                    if (!valid) {
+                        break;
+                    }
+                }
+                Thread.yield();
+            }
+        } catch (IOException e) {
+            _log.error("Error reading file: path={}", _file.getPath(), e);
+            stopThread();
+        }
+    }
+
+    public Map<String, String> readOnSplits() throws IOException {
+        Map<String, String> onSplits = Maps.newHashMap();
+
+        BufferedReader reader = null;
+        try {
+            reader = new BufferedReader(new FileReader(_file));
+            for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                line = line.trim();
+                if (line.isEmpty() || line.startsWith("#")) {
+                    continue;
+                }
+
+                String[] feature_treatment = line.split("\\s+");
+
+                if (feature_treatment.length != 2) {
+                    _log.info("Ignoring line since it does not have exactly two columns: " + line);
+                    continue;
+                }
+
+                onSplits.put(feature_treatment[0], feature_treatment[1]);
+                _log.info("100% of keys will see " + feature_treatment[1] + " for " + feature_treatment[0]);
+            }
+        } catch (FileNotFoundException e) {
+            _log.warn("There was no file named " + _file.getPath() + " found. " +
+                    "We created a split client that returns default treatments for all features for all of your users. " +
+                    "If you wish to return a specific treatment for a feature, enter the name of that feature name and " +
+                    "treatment name separated by whitespace in " + _file.getPath() +
+                    "; one pair per line. Empty lines or lines starting with '#' are considered comments", e);
+        } catch (IOException e) {
```

Comment:
This was a copy-paste from the code block in LocalhostSplitFactoryBuilder. Removed.